src/os/os.c

```diff
@@ -381,8 +396,30 @@ void vApplicationGetTimerTaskMemory(StaticTask_t **pxTimerTaskTCBBuffer,
     *ulTimerTaskStackSize = configMINIMAL_STACK_SIZE;
 }
 #endif // configSUPPORT_STATIC_ALLOCATION
-#endif // FreeRTOS
 
+/** Add a thread to the task list for tracking.
+ * @param task_new metadata for new task
+ */
+static void add_thread_to_task_list(TaskList *task_new)
+{
+    task_new->next = NULL;
+    vTaskSuspendAll();
+    if (taskList == NULL)
+    {
+        taskList = task_new;
+    }
+    else
+    {
+        TaskList *current = taskList;
+        while (current->next != NULL)
```

Comment:
what's the reason for needing to add the new task to the end of the task list? Why can't we add it to the beginning? It would be much simpler.